apps/addons/models.py

```diff
@@ -497,7 +494,14 @@ def delete(self, msg='', reason=''):
     @classmethod
     def initialize_addon_from_upload(cls, data, is_listed=True):
         fields = cls._meta.get_all_field_names()
+        # Reclaim GUID from deleted add-on.
+        try:
+            old_guid_addon = Addon.unfiltered.get(guid=data['guid'])
+            old_guid_addon.update(guid=None)
```

Comment:
ah, you're right. I'm just very tired, sorry for the noise. It's all good the way it is then.